src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,551 @@
+Q_GWPZ_DBM_PER_RBW = -80
+Q_FSS_CO_CHANNEL_DBM_PER_RBW = -129
+Q_FSS_BLOCKING_DBM_PER_RBW = -60
+Q_ESC_DBM_PER_RBW = -109
+
+# Default 1 dB pre IAP margin for all incumbent types
+MG_PPA_DB = 1
+MG_GWPZ_DB = 1
+MG_FSS_CO_CHANNEL_DB = 1
+MG_FSS_BLOCKING_DB = 1
+MG_ESC_DB = 1
+
+
+def iapPointConstraint(protection_point, channels, low_freq, high_freq, 
+                       grant_objects, fss_info, esc_antenna_info, region_type, 
+                       threshold, protection_ent_type, asas_interference, 
+                       aggregate_interference): 
+  """ Routine to compute aggregate interference(Ap and ASASp) from authorized grants 
+   
+  Aggregate interference is calculated over all the 5MHz channels.This routine 
+  is applicable for FSS Co-Channel and ESC Sensor protection points.Also for 
+  protection points within PPA and GWPZ protection areas. Post IAP grant EIRP
+  and interference contribution is calculated over each 5MHz channel
+
+  Args:
+    protection_point: A protection point as a tuple (longitude, latitude).
+    channels: List of tuples containing (low_freq, high_freq) of each 5MHz 
+              segment of the whole frequency range of protected entity
+    low_freq: low frequency of protection constraint (Hz)
+    high_freq: high frequency of protection constraint (Hz)
+    grant_objects: a list of grant requests, each one being a tuple 
+                   containing grant information
+    fss_info: contains fss point and antenna information
+    esc_antenna_info: contains information on ESC antenna height, azimuth, 
+                      gain and pattern gain
+    threshold: protection threshold (mW)
+    protection_ent_type: an enum member of class ProtectedEntityType 
+    asas_interference: object of class type AggregateInterferenceOutputFormat 
+    aggregate_interference: object of class type AggregateInterferenceOutputFormat 
+  Returns:
+    Updates asas_interference instance of class type AggregateInterferenceOutputFormat.
+    It is updated in the format of {latitude : {longitude : [ asas_p(mW/IAPBW),.. ]}}
+    and aggregate_interference instance of class type AggregateInterferenceOutputFormat.
+    It is updated in the format of {latitude : { longitude : [a_p(mW/IAPBW),..] }}
+  """
+  
+  # Get all the grants inside neighborhood of the protection entity
+  grants_inside = interf.findGrantsInsideNeighborhood(
+                    grant_objects, protection_point, protection_ent_type)
+    
+  if len(grants_inside) > 0:
+
+    for channel in channels:
+      # Set interference quota for protection_point, channel combination
+
+      # Calculate Roll of Attenuation for ESC Sensor Protection entity
+      if protection_ent_type is interf.ProtectedEntityType.ESC:
+        if channel[0] >= 3650.e6:
+          center_freq = (channel[0] + channel[1]) / 2
+          interference_quota = interf.dbToLinear(interf.linearToDb(threshold) - 
+                (2.5 + ((center_freq - interf.ESC_CH21_CF_HZ) / interf.MHZ)))
```

Comment:
Why is the interference_quota being updated here? The effect of the ESC filter roll-off should be accounted for when computing the interference power for each of the CBSD grants. 
This does not look right.

Reply:
James/   Sam to get back on this